python/MooseDocs/extensions/floats.py

```diff
@@ -107,6 +116,30 @@ def postTokenize(self, page, ast):
                     tokens.String(shortcut, content='{} {}'.format(prefix.title(), node['number']))
 
         page['counts'] = counts
+        page['floats'] = floats
+
+class FloatReferenceCommand(command.CommandComponent):
+    COMMAND = 'ref'
+    SUBCOMMAND = None
+    LABEL_RE = re.compile(r'((?P<filename>.*?\.md)#)?(?P<label>.+)', flags=re.UNICODE)
```

Comment:
Is this new syntax being explicitly tested? I see the impact on the changes to floats as a result of this (e.g., changing how `id` is represented in the syntax and the need for re-golding), but I'm not seeing testing of this new syntax. Apologies if I am missing it in the GitHub diff UI.

Reply:
Shoot, yeah, I need to test and document this.